contrib/cirrus/prebuild.sh

```diff
@@ -143,3 +143,10 @@ for testimg in "${TEST_IMGS[@]}"; do
     # in addition to "bad" (invalid) JSON.
     skopeo inspect --retry-times 5 "docker://$fqin" | jq -e . > /dev/null
 done
+
+msg "Checking [CI:MACHINE] only in draft PRs"
+if [[ "$CIRRUS_CHANGE_TITLE" =~ CI:MACHINE ]] && \
+       [[ "$CIRRUS_PR_DRAFT" != "true" ]] && [[ -n "$CIRRUS_PR" ]]; then
+  die "Machine-test only PRs must be in draft-mode to prevent merge of untested changes."
+  die "Risk of untested change merge.  Please remove [CI:MACHINE] from title."
+fi
```

Comment:
I think this gives a false sense of security. A PR can change from Draft to Ready with a single button tap, no need to rerun CI. It's therefore trivial to submit a Draft+CIMACHINE PR, then un-draft and have someone merge it.

My idea for fixing it isn't pretty: introduce a new intermediate "podman_all_machine_task", have `success` depend on this instead of individual `machine` tasks. The new `all` task will depend on all individual tasks **PLUS** one more, a one-bit task that will fail if `CI:MACHINE` is set. Diagram thousand words etc.
![cirrus-map-pr20946-cropped](https://github.com/containers/podman/assets/265639/186a7576-e257-41ec-ac00-9aa760150f55)

The idea is that this cannot be bypassed with a single button click.

Reply:
My other thought here is: It's too hard, just remove the machine dependencies so the tests always start earlier.  Esp. in the case of the Mac tests, we're paying for those machines w/n they're actually used.